tools/ci/tc/decision.py

```diff
@@ -121,24 +121,31 @@ def get_fetch_rev(event):
     is_pr, _ = get_triggers(event)
     if is_pr:
         # Try to get the actual rev so that all non-decision tasks are pinned to that
-        ref = "refs/pull/%s/merge" % event["pull_request"]["number"]
-        try:
-            output = subprocess.check_output(["git", "ls-remote", "origin", ref])
-        except subprocess.CalledProcessError:
-            import traceback
-            logger.error(traceback.format_exc())
-            logger.error("Failed to get merge commit sha1")
-            return ref, None
-        if not output:
-            logger.error("Failed to get merge commit")
-            return ref, None
-        return ref, output.split()[0]
+        rv = ["refs/pull/%s/merge" % event["pull_request"]["number"]]
+        for ref_type in ["head", "merge"]:
+            ref = "refs/pull/%s/%s" % (event["pull_request"]["number"], ref_type)
+            sha = None
+            try:
+                output = subprocess.check_output(["git", "ls-remote", "origin", ref])
+            except subprocess.CalledProcessError:
+                import traceback
+                logger.error(traceback.format_exc())
+                logger.error("Failed to get commit sha1 for %s" % ref)
+            else:
+                if not output:
+                    logger.error("Failed to get commit for %s" % ref)
+                else:
+                    sha = output.split()[0]
+            rv.append(sha)
+        rv = tuple(rv)
     else:
-        return event["ref"], event["after"]
+        rv = (event["ref"], event["after"], None)
```

Comment:
Can you add a brief comment noting that there is no 'merge' SHA for a branch push?